### PR TITLE
[Data objects] Objects of wrong folder displayed

### LINKED ISSUE #6751: [Data objects] Objects of wrong folder displayed
Steps to reproduce:
In Pimcore backend:
1. Create folder `_0` under `/`
2. Create folder `00` under `/`
3. Create object `A` in `/_0`
4. Create object `B` in `/00`
5. Open folder `_0` --> you will see both objects in the grid although actually only `A` is in folder `/_0` while `B` is not.

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #6751

### CODE REVIEW

File: lib/Db/PimcoreExtensionsTrait.php [hunk 1]
```diff
@@ -635,4 +635,15 @@ public function selectAndDeleteWhere($table, $idColumn = 'id', $where = '')
...
     }
+
+
+
+    /**
+     * @param string $like
+     * @return string
+     */
+    public function escapeLike(string $like) : string {
+        return str_replace("_", '\\_', $like);
```
Comment (brusch) [inline lib/Db/PimcoreExtensionsTrait.php:646]:
```suggestion
        return str_replace(['_','%'], ['\\_', '\\%'], $like);
```

File: models/Asset/Dao.php [hunk 2]
```diff
@@ -158,7 +159,7 @@ public function updateWorkspaces()
     public function updateChildPaths($oldPath)
     {
         //get assets to empty their cache
-        $assets = $this->db->fetchCol('SELECT id FROM assets WHERE path LIKE ' . $this->db->quote($oldPath . '%'));
+        $assets = $this->db->fetchCol('SELECT id FROM assets WHERE path LIKE ' . $this->db->quote(Db::get()->escapeLike($oldPath . '%')));
```
Comment (brusch) [inline models/Asset/Dao.php:162]: Whenever possible we shouldn't use the static getter, better use `$this->db->escapeLike()` instead if possible.

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 3]
```diff
@@ -630,7 +630,7 @@ public function deleteAction(Request $request)
             $parentAsset = Asset::getById($request->get('id'));
 
             $list = new Asset\Listing();
-            $list->setCondition('path LIKE ?', [$parentAsset->getRealFullPath() . '/%']);
+            $list->setCondition('path LIKE ?', [Db::get()->escapeLike($parentAsset->getRealFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:633]: I'm just thinking, maybe we should provide `escapeLike()` on the abstract listing class as well, as it is used quite often and we should avoid the static calls? wdyt?

**Comment by dvesh3** [comment]
@weisswurstkanone please check, tests are failing. thanks!

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 4]
```diff
@@ -1918,7 +1918,7 @@ public function downloadAsZipJobsAction(Request $request)
                 //add a condition if id numbers are specified
                 $conditionFilters[] = 'id IN (' . implode(',', $quotedSelectedIds) . ')';
             }
-            $conditionFilters[] = 'path LIKE ' . $db->quote($parentPath . '/%') .' AND type != ' . $db->quote('folder');
+            $conditionFilters[] = 'path LIKE ' . $db->quote(Db::get()->escapeLike($parentPath) . '/%') .' AND type != ' . $db->quote('folder');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:1921]:
```suggestion
            $conditionFilters[] = 'path LIKE ' . $db->quote($db->escapeLike($parentPath) . '/%') .' AND type != ' . $db->quote('folder');
```

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 5]
```diff
@@ -1999,7 +1999,7 @@ public function downloadAsZipAddFilesAction(Request $request)
                     //add a condition if id numbers are specified
                     $conditionFilters[] = 'id IN (' . implode(',', $selectedIds) . ')';
                 }
-                $conditionFilters[] = "type != 'folder' AND path LIKE " . $db->quote($parentPath . '/%');
+                $conditionFilters[] = "type != 'folder' AND path LIKE " . $db->quote(Db::get()->escapeLike($parentPath) . '/%');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:2002]:
```suggestion
                $conditionFilters[] = "type != 'folder' AND path LIKE " . $db->quote($db->escapeLike($parentPath) . '/%');
```

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php [hunk 6]
```diff
@@ -209,7 +210,7 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
                         $tag = Element\Tag::getById($tagId);
                         if ($tag) {
                             $tagPath = $tag->getFullIdPath();
-                            $conditionParts[] = 'id IN (SELECT cId FROM tags_assignment INNER JOIN tags ON tags.id = tags_assignment.tagid WHERE ctype = ' . $db->quote($type) . ' AND (id = ' . intval($tagId) . ' OR idPath LIKE ' . $db->quote($tagPath . '%') . '))';
+                            $conditionParts[] = 'id IN (SELECT cId FROM tags_assignment INNER JOIN tags ON tags.id = tags_assignment.tagid WHERE ctype = ' . $db->quote($type) . ' AND (id = ' . intval($tagId) . ' OR idPath LIKE ' . $db->quote(Db::get()->escapeLike($tagPath) . '%') . '))';
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Searchadmin/SearchController.php:213]:
```suggestion
                            $conditionParts[] = 'id IN (SELECT cId FROM tags_assignment INNER JOIN tags ON tags.id = tags_assignment.tagid WHERE ctype = ' . $db->quote($type) . ' AND (id = ' . intval($tagId) . ' OR idPath LIKE ' . $db->quote($db->escapeLike($tagPath) . '%') . '))';
```

File: bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractBatchProcessingWorker.php [hunk 7]
```diff
@@ -381,7 +382,7 @@ public function fillupPreparationQueue(IndexableInterface $object)
         if ($object instanceof Concrete) {
 
             //need check, if there are sub objects because update on empty result set is too slow
-            $objects = $this->db->fetchCol('SELECT o_id FROM objects WHERE o_path LIKE ?', [$object->getFullPath() . '/%']);
+            $objects = $this->db->fetchCol('SELECT o_id FROM objects WHERE o_path LIKE ?', [Db::get()->escapeLike($object->getFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractBatchProcessingWorker.php:385]:
```suggestion
            $objects = $this->db->fetchCol('SELECT o_id FROM objects WHERE o_path LIKE ?', [$this->db->escapeLike($object->getFullPath()) . '/%']);
```

File: models/Document/Printcontainer/Dao.php [hunk 8]
```diff
@@ -31,6 +32,6 @@ public function getLastedChildModificationDate()
     {
         $path = $this->model->getFullPath();
 
-        return $this->db->fetchOne('SELECT modificationDate FROM documents WHERE path LIKE ? ORDER BY modificationDate DESC LIMIT 0,1', [$path . '%']);
+        return $this->db->fetchOne('SELECT modificationDate FROM documents WHERE path LIKE ? ORDER BY modificationDate DESC LIMIT 0,1', [Db::get()->escapeLike($path) . '%']);
```
Comment (brusch) [inline models/Document/Printcontainer/Dao.php:35]:
```suggestion
        return $this->db->fetchOne('SELECT modificationDate FROM documents WHERE path LIKE ? ORDER BY modificationDate DESC LIMIT 0,1', [$this->db->escapeLike($path) . '%']);
```

File: models/Document/Service/Dao.php [hunk 9]
```diff
@@ -38,7 +39,7 @@ public function getDocumentIdByPrettyUrlInSite(Site $site, $path)
             'SELECT documents.id FROM documents
             LEFT JOIN documents_page ON documents.id = documents_page.id
             WHERE documents.path LIKE ? AND documents_page.prettyUrl = ?',
-        [$site->getRootPath() . '/%', rtrim($path, '/')]
+        [Db::get()->escapeLike($site->getRootPath()) . '/%', rtrim($path, '/')]
```
Comment (brusch) [inline models/Document/Service/Dao.php:42]:
```suggestion
        [$this->db->escapeLike($site->getRootPath()) . '/%', rtrim($path, '/')]
```

File: models/Element/PermissionChecker.php [hunk 10]
```diff
@@ -119,7 +119,7 @@ public static function check(ElementInterface $element, $users)
                                 ',',
                                 $userIds
                             ).') AND list = 1 LIMIT 1',
-                            $path.'%'
+                            Db::get()->escapeLike($path) .'%'
```
Comment (brusch) [inline models/Element/PermissionChecker.php:122]:
```suggestion
                            $db->escapeLike($path) .'%'
```

File: models/Element/Tag/Dao.php [hunk 11]
```diff
@@ -79,7 +80,7 @@ public function save()
 
             //check for id-path and update it, if path has changed -> update all other tags that have idPath == idPath/id
             if ($originalIdPath && $originalIdPath != $this->model->getIdPath()) {
-                $this->db->query('UPDATE tags SET idPath = REPLACE(idPath, ?, ?)  WHERE idPath LIKE ?;', [$originalIdPath, $this->model->getIdPath(), $originalIdPath . $this->model->getId() . '/%']);
+                $this->db->query('UPDATE tags SET idPath = REPLACE(idPath, ?, ?)  WHERE idPath LIKE ?;', [$originalIdPath, $this->model->getIdPath(), Db::get()->escapeLike($originalIdPath) . $this->model->getId() . '/%']);
```
Comment (brusch) [inline models/Element/Tag/Dao.php:83]:
```suggestion
                $this->db->query('UPDATE tags SET idPath = REPLACE(idPath, ?, ?)  WHERE idPath LIKE ?;', [$originalIdPath, $this->model->getIdPath(), $this->db->escapeLike($originalIdPath) . $this->model->getId() . '/%']);
```

File: models/Element/Tag/Dao.php [hunk 12]
```diff
@@ -101,10 +102,10 @@ public function delete()
         $this->db->beginTransaction();
         try {
             $this->db->delete('tags_assignment', ['tagid' => $this->model->getId()]);
-            $this->db->deleteWhere('tags_assignment', $this->db->quoteInto('tagid IN (SELECT id FROM tags WHERE idPath LIKE ?)', $this->model->getIdPath() . $this->model->getId() . '/%'));
+            $this->db->deleteWhere('tags_assignment', $this->db->quoteInto('tagid IN (SELECT id FROM tags WHERE idPath LIKE ?)', Db::get()->escapeLike($this->model->getIdPath()) . $this->model->getId() . '/%'));
```
Comment (brusch) [inline models/Element/Tag/Dao.php:105]:
```suggestion
            $this->db->deleteWhere('tags_assignment', $this->db->quoteInto('tagid IN (SELECT id FROM tags WHERE idPath LIKE ?)', $this->db->escapeLike($this->model->getIdPath()) . $this->model->getId() . '/%'));
```

File: models/Element/Tag/Dao.php [hunk 13]
```diff
@@ -101,10 +102,10 @@ public function delete()
...
 
             $this->db->delete('tags', ['id' => $this->model->getId()]);
-            $this->db->deleteWhere('tags', $this->db->quoteInto('idPath LIKE ?', $this->model->getIdPath() . $this->model->getId() . '/%'));
+            $this->db->deleteWhere('tags', $this->db->quoteInto('idPath LIKE ?', Db::get()->escapeLike($this->model->getIdPath()) . $this->model->getId() . '/%'));
```
Comment (brusch) [inline models/Element/Tag/Dao.php:108]:
```suggestion
            $this->db->deleteWhere('tags', $this->db->quoteInto('idPath LIKE ?', $this->db->escapeLike($this->model->getIdPath()) . $this->model->getId() . '/%'));
```

File: models/Element/Tag/Dao.php [hunk 14]
```diff
@@ -255,7 +256,7 @@ public function getElementsForTag(
             $select->where(
                 '(' .
                 $this->db->quoteInto('tags_assignment.tagid = ?', $tag->getId()) . ' OR ' .
-                $this->db->quoteInto('tags.idPath LIKE ?', $tag->getFullIdPath() . '%') . ')'
+                $this->db->quoteInto('tags.idPath LIKE ?', Db::get()->escapeLike($tag->getFullIdPath()) . '%' . ')')
```
Comment (brusch) [inline models/Element/Tag/Dao.php:259]:
```suggestion
                $this->db->quoteInto('tags.idPath LIKE ?', $this->db->escapeLike($tag->getFullIdPath()) . '%' . ')')
```

File: bundles/AdminBundle/Controller/Admin/AssetController.php (hunk 3 above)
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:633]:
```suggestion
            $list->setCondition('path LIKE ?', [$list->escapeLike($parentAsset->getRealFullPath()) . '/%']);
```

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 15]
```diff
@@ -699,7 +699,7 @@ protected function getTreeNodeConfig($element)
 
             $folderThumbs = [];
             $children = new Asset\Listing();
-            $children->setCondition('path LIKE ?', [$asset->getRealFullPath() . '/%']);
+            $children->setCondition('path LIKE ?', [Db::get()->escapeLike($asset->getRealFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:702]:
```suggestion
            $children->setCondition('path LIKE ?', [$children->escapeLike($asset->getRealFullPath()) . '/%']);
```

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 16]
```diff
@@ -1666,7 +1666,7 @@ public function getFolderContentPreviewAction(Request $request, EventDispatcherI
 
         $conditionFilters = [];
         $list = new Asset\Listing();
-        $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() == '/' ? "'/%'" : $list->quote($folder->getRealFullPath() . '/%')) ." AND type != 'folder'";
+        $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() == '/' ? "'/%'" : $list->quote(Db::get()->escapeLike($folder->getRealFullPath()) . '/%')) ." AND type != 'folder'";
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:1669]:
```suggestion
        $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() == '/' ? "'/%'" : $list->quote($list->escapeLike($folder->getRealFullPath()) . '/%')) ." AND type != 'folder'";
```

File: bundles/AdminBundle/Controller/Admin/AssetController.php [hunk 17]
```diff
@@ -1773,7 +1773,7 @@ public function copyInfoAction(Request $request)
             if ($asset->hasChildren()) {
                 // get amount of children
                 $list = new Asset\Listing();
-                $list->setCondition('path LIKE ?', [$asset->getRealFullPath() . '/%']);
+                $list->setCondition('path LIKE ?', [Db::get()->escapeLike($asset->getRealFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/AssetController.php:1776]:
```suggestion
                $list->setCondition('path LIKE ?', [$list->escapeLike($asset->getRealFullPath()) . '/%']);
```

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 18]
```diff
@@ -948,7 +948,7 @@ public function deleteAction(Request $request)
             $parentObject = DataObject::getById($request->get('id'));
 
             $list = new DataObject\Listing();
-            $list->setCondition('o_path LIKE ' . $list->quote($parentObject->getRealFullPath() . '/%'));
+            $list->setCondition('o_path LIKE ' . $list->quote(Db::get()->escapeLike($parentObject->getRealFullPath()) . '/%'));
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:951]:
```suggestion
            $list->setCondition('o_path LIKE ' . $list->quote($list->escapeLike($parentObject->getRealFullPath()) . '/%'));
```

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 19]
```diff
@@ -1869,7 +1869,7 @@ public function copyInfoAction(Request $request)
             if ($object->hasChildren([DataObject\AbstractObject::OBJECT_TYPE_OBJECT, DataObject\AbstractObject::OBJECT_TYPE_FOLDER, DataObject\AbstractObject::OBJECT_TYPE_VARIANT])) {
                 // get amount of children
                 $list = new DataObject\Listing();
-                $list->setCondition('o_path LIKE ' . $list->quote($object->getRealFullPath() . '/%'));
+                $list->setCondition('o_path LIKE ' . $list->quote(Db::get()->escapeLike($object->getRealFullPath()) . '/%'));
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php:1872]:
```suggestion
                $list->setCondition('o_path LIKE ' . $list->quote($list->escapeLike($object->getRealFullPath()) . '/%'));
```

File: bundles/AdminBundle/Controller/Admin/Document/DocumentController.php [hunk 20]
```diff
@@ -383,7 +383,7 @@ public function deleteAction(Request $request)
             $parentDocument = Document::getById($request->get('id'));
 
             $list = new Document\Listing();
-            $list->setCondition('path LIKE ?', [$parentDocument->getRealFullPath() . '/%']);
+            $list->setCondition('path LIKE ?', [Db::get()->escapeLike($parentDocument->getRealFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/Document/DocumentController.php:386]:
```suggestion
            $list->setCondition('path LIKE ?', [$list->escapeLike($parentDocument->getRealFullPath()) . '/%']);
```

File: bundles/AdminBundle/Controller/Admin/Document/DocumentController.php [hunk 21]
```diff
@@ -552,7 +552,7 @@ private function createRedirectForFormerPath(Request $request, Document $documen
                     if ($document->hasChildren()) {
                         $list = new Document\Listing();
                         $list->setCondition('path LIKE :path', [
-                            'path' => $document->getRealFullPath() . '/%'
+                            'path' => Db::get()->escapeLike($document->getRealFullPath()) . '/%'
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/Document/DocumentController.php:555]:
```suggestion
                            'path' => $list->escapeLike($document->getRealFullPath()) . '/%'
```

File: bundles/AdminBundle/Controller/Admin/Document/DocumentController.php [hunk 22]
```diff
@@ -869,7 +869,7 @@ public function copyInfoAction(Request $request)
             if ($document->hasChildren()) {
                 // get amount of childs
                 $list = new Document\Listing();
-                $list->setCondition('path LIKE ?', [$document->getRealFullPath() . '/%']);
+                $list->setCondition('path LIKE ?', [Db::get()->escapeLike($document->getRealFullPath()) . '/%']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/Document/DocumentController.php:872]:
```suggestion
                $list->setCondition('path LIKE ?', [$list->escapeLike($document->getRealFullPath()) . '/%']);
```

File: bundles/AdminBundle/Controller/Admin/Document/DocumentController.php [hunk 23]
```diff
@@ -1333,7 +1333,7 @@ public function seopanelTreeAction(
                 // only display document if listing is allowed for the current user
                 if ($childDocument->isAllowed('list')) {
                     $list = new Document\Listing();
-                    $list->setCondition('path LIKE ? and type = ?', [$childDocument->getRealFullPath() . '/%', 'page']);
+                    $list->setCondition('path LIKE ? and type = ?', [Db::get()->escapeLike($childDocument->getRealFullPath() ). '/%', 'page']);
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/Document/DocumentController.php:1336]:
```suggestion
                    $list->setCondition('path LIKE ? and type = ?', [$list->escapeLike($childDocument->getRealFullPath() ). '/%', 'page']);
```

File: bundles/AdminBundle/Controller/Admin/RecyclebinController.php [hunk 24]
```diff
@@ -59,7 +60,7 @@ public function listAction(Request $request)
             $conditionFilters = [];
 
             if ($request->get('filterFullText')) {
-                $conditionFilters[] = 'path LIKE ' . $list->quote('%'.$request->get('filterFullText').'%');
+                $conditionFilters[] = 'path LIKE ' . $list->quote('%'. Db::get()->escapeLike($request->get('filterFullText')) .'%');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/RecyclebinController.php:63]:
```suggestion
                $conditionFilters[] = 'path LIKE ' . $list->quote('%'. $list->escapeLike($request->get('filterFullText')) .'%');
```

File: bundles/AdminBundle/Controller/Admin/RecyclebinController.php [hunk 25]
```diff
@@ -174,7 +175,7 @@ public function addAction(Request $request)
                 $baseClass = Element\Service::getBaseClassNameForElement($type);
                 $listClass = '\\Pimcore\\Model\\' . $baseClass . '\\Listing';
                 $list = new $listClass();
-                $list->setCondition((($type == 'object') ? 'o_' : '') . 'path LIKE ' . $list->quote($element->getRealFullPath() . '/%'));
+                $list->setCondition((($type == 'object') ? 'o_' : '') . 'path LIKE ' . $list->quote(Db::get()->escapeLike($element->getRealFullPath()) . '/%'));
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/RecyclebinController.php:178]:
```suggestion
                $list->setCondition((($type == 'object') ? 'o_' : '') . 'path LIKE ' . $list->quote($list->escapeLike($element->getRealFullPath()) . '/%'));
```

File: bundles/AdminBundle/Controller/Admin/TagsController.php [hunk 26]
```diff
@@ -327,7 +328,7 @@ private function getSubObjectIds(\Pimcore\Model\DataObject\AbstractObject $objec
              )';
         }
 
-        $childsList->setCondition($condition, $object->getRealFullPath() . '/%');
+        $childsList->setCondition($condition, Db::get()->escapeLike($object->getRealFullPath()) . '/%');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/TagsController.php:331]:
```suggestion
        $childsList->setCondition($condition, $childsList->escapeLike($object->getRealFullPath()) . '/%');
```

File: bundles/AdminBundle/Controller/Admin/TagsController.php [hunk 27]
```diff
@@ -359,7 +360,7 @@ private function getSubAssetIds(\Pimcore\Model\Asset $asset, EventDispatcherInte
             )';
         }
 
-        $childsList->setCondition($condition, $asset->getRealFullPath() . '/%');
+        $childsList->setCondition($condition, Db::get()->escapeLike($asset->getRealFullPath()) . '/%');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/TagsController.php:363]:
```suggestion
        $childsList->setCondition($condition, $childsList->escapeLike($asset->getRealFullPath()) . '/%');
```

File: bundles/AdminBundle/Controller/Admin/TagsController.php [hunk 28]
```diff
@@ -391,7 +392,7 @@ private function getSubDocumentIds(\Pimcore\Model\Document $document, EventDispa
             )';
         }
 
-        $childsList->setCondition($condition, $document->getRealFullPath() . '/%');
+        $childsList->setCondition($condition, Db::get()->escapeLike($document->getRealFullPath()) . '/%');
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/TagsController.php:395]:
```suggestion
        $childsList->setCondition($condition, $childsList->escapeLike($document->getRealFullPath()) . '/%');
```

File: bundles/AdminBundle/Controller/Admin/TranslationController.php [hunk 29]
```diff
@@ -714,7 +715,7 @@ public function contentExportJobsAction(Request $request)
                     }
                     $list->setCondition(
                         ($el instanceof DataObject\AbstractObject ? 'o_' : '') . 'path LIKE ?',
-                        [$el->getRealFullPath() . ($el->getRealFullPath() != '/' ? '/' : '') . '%']
+                        [Db::get()->escapeLike($el->getRealFullPath() . ($el->getRealFullPath() != '/' ? '/' : '')) . '%']
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/TranslationController.php:718]:
```suggestion
                        [$list->escapeLike($el->getRealFullPath() . ($el->getRealFullPath() != '/' ? '/' : '')) . '%']
```

File: bundles/AdminBundle/Helper/GridHelperService.php [hunk 30]
```diff
@@ -553,7 +553,7 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
             $conditionFilters[] = 'o_parentId = ' . $folder->getId();
         } else {
             $quotedPath = $list->quote($folder->getRealFullPath());
-            $quotedWildcardPath = $list->quote(str_replace('//', '/', $folder->getRealFullPath() . '/') . '%');
+            $quotedWildcardPath = $list->quote(Db::get()->escapeLike(str_replace('//', '/', $folder->getRealFullPath() . '/')) . '%');
```
Comment (brusch) [inline bundles/AdminBundle/Helper/GridHelperService.php:556]:
```suggestion
            $quotedWildcardPath = $list->quote($list->escapeLike(str_replace('//', '/', $folder->getRealFullPath() . '/')) . '%');
```

File: bundles/AdminBundle/Helper/GridHelperService.php [hunk 31]
```diff
@@ -691,7 +691,7 @@ public function prepareAssetListingForGrid($allParams, $adminUser)
         if (isset($allParams['only_direct_children']) && $allParams['only_direct_children'] == 'true') {
             $conditionFilters[] = 'parentId = ' . $folder->getId();
         } else {
-            $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() === '/' ? "'/%'" : $list->quote($folder->getRealFullPath() . '/%'));
+            $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() === '/' ? "'/%'" : $list->quote(Db::get()->escapeLike($folder->getRealFullPath()) . '/%'));
```
Comment (brusch) [inline bundles/AdminBundle/Helper/GridHelperService.php:694]:
```suggestion
            $conditionFilters[] = 'path LIKE ' . ($folder->getRealFullPath() === '/' ? "'/%'" : $list->quote($list->escapeLike($folder->getRealFullPath()) . '/%'));
```

**Pull request merged**